### ISSUE TITLE
Strict pyright inconsistently complains about partially unknown generic types in instance check tuples

### ISSUE THREAD

**Issue by FasterSpeeding** [microsoft/pyright#2618]
**Describe the bug**
Pyright complains about generic types being "partially unknown" in instance checks when multiple types are passed in a tuple and they're being accessed as a attributes (e.g. on a module or class) since this case specifically triggers reportUnknownMemberType.

This deviates from how the types would be treated if only one of them is passed directly to the instance check (rather than in a tuple) and if they were passed as flat variables (rather than module attributes) in a tuple since both of these other two similar cases seem to be special cased to ignore the fact that they're partially unknown.

**To Reproduce**
1. Make sure pyright is set to strict.
2. type check the example code.
3. Watch it give out the following errors.
```py
C:\Users\Lucina\PycharmProjects\tanjun\test.py
  C:\Users\Lucina\PycharmProjects\tanjun\test.py:23:25 - error: Type of "Foo" is partially unknown
    Type of "Foo" is "Type[Foo[Unknown]]" (reportUnknownMemberType)
  C:\Users\Lucina\PycharmProjects\tanjun\test.py:23:37 - error: Type of "Bar" is partially unknown
    Type of "Bar" is "Type[Bar[Unknown]]" (reportUnknownMemberType)
2 errors, 0 warnings, 0 infos
```

**Expected behavior**
Like the other two cases previously described, this case of the partially unknown generics being passed to an instance check should be ignored as you cannot make them fully known without breaking the instance check or casting the classes.

**Screenshots or Code**
```py
import typing

# Defining _T in Module's namespace leads to pyright rightfully so complaining
# about it not being defined when used in return types but if Module was an
# actual module it would be defined in its global namespace.
_T = typing.TypeVar("_T")


# Here I've used a class to mock the namespace of a module for the sake of
# brevity but this (including _T) could otherwise be moved to a separate module
# and accessed as attributes on the imported module to hit the same error case.
class Module:
    class Foo(typing.Generic[_T]):
        def get_foo(self) -> _T:
            raise NotImplementedError

    class Bar(typing.Generic[_T]):
        def get_foo(self) -> _T:
            raise NotImplementedError


def too_foo(foo: typing.Union[Module.Foo[int], Module.Bar[int], int]) -> int:
    if isinstance(foo, (Module.Foo, Module.Bar)):
        return foo.get_foo()

    return foo
```

**VS Code extension or command-line**
Pyright v1.1.190 as a CLI tool.

**Additional context**
I was originally going to open this as a discussion but the more I looked into this the more it looked like a missing special case so I decided to go straight to a bug report.

**Comment by erictraut** [comment]
Yes, there's already an ugly special case to suppress the "partially unknown type" check when a member access expression is used directly as an argument to a call. You're wrapping a member access expression in a tuple expression, so the special case didn't apply. I've expanded the special case to handle other expression forms.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.191, which I just published. It will also be included in the next release of pylance.
